src/sosdec.jl

```diff
@@ -70,3 +69,83 @@ function Base.isapprox(p::SOSDecomposition, q::SOSDecomposition; kwargs...)
         MultivariateMoments.permcomp((i, j) -> isapprox(p.ps[i], q.ps[j]; kwargs...), m)
     end
 end
+
+function Base.promote_rule(::Type{SOSDecomposition{T1, PT1}}, ::Type{SOSDecomposition{T2, PT2}}) where {T1, T2, PT1<:MP.APL{T1}, PT2<:MP.APL{T2}}
+	return SOSDecomposition{promote_type(T1, T2), promote_type(PT1, PT2)} 
+end
+
+function Base.convert(::Type{SOSDecomposition{T, PT}}, p::SOSDecomposition) where {T, PT}
+	return SOSDecomposition(convert(Vector{PT}, p.ps))
+end
+
+function MP.polynomial(decomp::SOSDecomposition)
+    return sum(decomp.ps.^2)
+end
+function MP.polynomial(decomp::SOSDecomposition, T::Type)
+    return MP.polynomial(MP.polynomial(decomp), T)
+end
+
+"""
+	function sos_decomposition(cref::JuMP.ConstraintRef, ranktol::Float64, dec::MultivariateMoments.lowrankchol)
+
+Returns representation as a sum of squares.
+"""
+function sos_decomposition(cref::JuMP.ConstraintRef, args...)
+	return SOSDecomposition(gram_matrix(cref), args...)
+end
+
+
+"""
+	struct SOSDecompositionWithDomain{T, PT, S}
+
+Represends SOSDecomposition on a basic semi-algebraic domain.
```

Comment:
You should say that `sosj[j]` corresponds to the multiplier for the `j`th inequality of `domain`